src/main/java/net/glowstone/block/blocktype/BlockCauldron.java

```diff
@@ -61,36 +75,74 @@ public boolean blockInteract(GlowPlayer player, GlowBlock block, BlockFace face,
         }
     }
 
-    private void fillCauldron(GlowPlayer player, GlowBlock block) {
+    private void emptyBucket(GlowPlayer player, GlowBlock block) {
         // fired when a player fills the cauldron using a water bucket
-        if (block.getData() < LEVEL_FULL) {
-            if (!setCauldronLevel(block, LEVEL_FULL, player,
-                    CauldronLevelChangeEvent.ChangeReason.BUCKET_EMPTY)) {
-                return;
-            }
-            if (player.getGameMode() != GameMode.CREATIVE) {
-                player.getItemInHand().setType(Material.BUCKET);
+        if (block.getData() == LEVEL_FULL) {
+            return;
+        }
+        if (!setCauldronLevel(block, LEVEL_FULL, player,
+                CauldronLevelChangeEvent.ChangeReason.BUCKET_EMPTY)) {
+            return;
+        }
+        if (player.getGameMode() != GameMode.CREATIVE) {
+            player.getItemInHand().setType(Material.BUCKET);
+        }
+
+    }
+
+    private void fillBucket(GlowPlayer player, GlowBlock block) {
+        // fired when a player fills an empty bucket from the cauldron
+        if (block.getData() != LEVEL_FULL) {
+            return;
+        }
+        if (!setCauldronLevel(block, LEVEL_EMPTY, player,
+                CauldronLevelChangeEvent.ChangeReason.BUCKET_FILL)) {
+            return;
+        }
+        if (player.getGameMode() != GameMode.CREATIVE) {
+            Map<Integer, ItemStack> dummyInventory = player.getInventory()
```

Comment:
Can we rename the variable to `drops`, for consistency?

Reply:
Of course! Will comply in next commit.